### PR TITLE
Workflow deps

### PR DESCRIPTION
**Pull request by jmhauck**

### CODE REVIEW

File: packages/common/src/workflowHelpers.ts [hunk 1]
```diff
@@ -310,3 +310,22 @@ export function updateWorkflowTemplateIds(
...
+ * Get the various dependencies from the workflow item
+ *
+ * @param template The IItemTemplate of the workflow item
+ *
+ */
+export function getWorkflowDependencies(template: interfaces.IItemTemplate): void {
+  const keys = ["viewSchema", "workflowLocations", "workflowSchema"];
+  keys.forEach((k) => {
+    const data = template.data;
+    if (data && Object.keys(data).indexOf(k) > -1) {
```
Comment (MikeTschudi) [inline packages/common/src/workflowHelpers.ts:324]: How about moving the test for `data` existing out of the loop?

File: packages/common/src/workflowHelpers.ts [hunk 2]
```diff
@@ -310,3 +310,22 @@ export function updateWorkflowTemplateIds(
   }
   return templates;
 }
+
+/**
+ * Get the various dependencies from the workflow item
+ *
+ * @param template The IItemTemplate of the workflow item
```
Comment (MikeTschudi) [inline packages/common/src/workflowHelpers.ts:317]: "...; the ids of found dependencies are added to the template's list of dependencies"

**Review by MikeTschudi — APPROVED** [review]

**Comment by jmhauck** [comment]
hey @MikeTschudi I needed to add a bit more and forgot to switch branches if you could take a look when you get a chance

File: packages/creator/src/createItemTemplate.ts [hunk 3]
```diff
@@ -549,24 +551,41 @@ export function _templatizeResources(
...
               ids.forEach((id) => {
+                idLookup.push(id);
                 if (verifiedIds.indexOf(id) === -1) {
                   verifiedIds.push(id);
-                  // templatize the itemId--but only once per unique id
-                  const regEx = new RegExp(id, "gm");
-                  dataString = dataString.replace(regEx, "{{" + id + ".itemId}}");
+                  promises.push(isItem(id, srcAuthentication));
+                  promises.push(isGroup(id, srcAuthentication));
+                }
```
Comment (MikeTschudi) [inline packages/creator/src/createItemTemplate.ts:562]: Interesting improvement over checking first for item and then for group or vice versa!

**Review by MikeTschudi — APPROVED** [review]

**Pull request merged**